### ISSUE TITLE
Error while writing unindexed boolean

### ISSUE THREAD

**Issue by jxerome** [strapdata/elassandra#135]
**Elassandra version**: [ 5.5.0.3, 5.5.0.4, 5.5.0.5 ]

**JVM version**: Java HotSpot(TM) 64-Bit Server VM (build 25.151-b12, mixed mode)

**OS version**: Linux 4.10.0-38-generic

**Description of the problem including expected versus actual behavior**:

**Steps to reproduce**:
 1. Create an index with a not indexed boolean
```
PUT elassandra
{
  "mappings" : {
    "fan" : {
      "dynamic" : "strict",
      "properties" : {
        "name" : {
          "type" : "text"
        },
        "like_elassandra" : {
          "index" : false,
          "type" : "boolean"
        }
      }
    }
  }
}

```

 2. Index a document
```
PUT elassandra/fan/jxerome
{
    "name": "jxerome",
    "like_elassandra" : true
}
```

 3. The last query fail with this message:
```
{
    "error": {
        "root_cause": [
            {
                "type": "remote_transport_exception",
                "reason": "[127.0.0.1][127.0.0.1:9300][indices:data/write/bulk[s][p]]"
            }
        ],
        "type": "write_failure_exception",
        "reason": "Operation failed - received 0 responses and 1 failures"
    },
    "status": 500
}
```

**Provide logs (if relevant)**:
```
2017-11-09 17:22:35,093 ERROR [MutationStage-1] org.apache.cassandra.service.StorageProxy$8.runMayThrow(StorageProxy.java:1414) Failed to apply mutation locally : {}
java.lang.RuntimeException: it doesn't make sense to have a field that is neither indexed nor stored for ks: elassandra, table: fan
	at org.apache.cassandra.db.ColumnFamilyStore.apply(ColumnFamilyStore.java:1357)
	at org.apache.cassandra.db.Keyspace.applyInternal(Keyspace.java:623)
	at org.apache.cassandra.db.Keyspace.apply(Keyspace.java:467)
	at org.apache.cassandra.db.Mutation.apply(Mutation.java:227)
	at org.apache.cassandra.db.Mutation.apply(Mutation.java:232)
	at org.apache.cassandra.db.Mutation.apply(Mutation.java:241)
	at org.apache.cassandra.service.StorageProxy$8.runMayThrow(StorageProxy.java:1408)
	at org.apache.cassandra.service.StorageProxy$LocalMutationRunnable.run(StorageProxy.java:2647)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at org.apache.cassandra.concurrent.AbstractLocalAwareExecutorService$FutureTask.run(AbstractLocalAwareExecutorService.java:162)
	at org.apache.cassandra.concurrent.SEPWorker.run(SEPWorker.java:109)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.lang.IllegalArgumentException: it doesn't make sense to have a field that is neither indexed nor stored
	at org.apache.lucene.document.Field.<init>(Field.java:249)
	at org.elasticsearch.index.mapper.BooleanFieldMapper.createField(BooleanFieldMapper.java:277)
	at org.elassandra.index.ElasticSecondaryIndex$Context.addField(ElasticSecondaryIndex.java:404)
	at org.elassandra.index.ElasticSecondaryIndex$Context.addField(ElasticSecondaryIndex.java:376)
	at org.elassandra.index.ElasticSecondaryIndex$ImmutableMappingInfo$RowcumentIndexer$Rowcument.buildContext(ElasticSecondaryIndex.java:1805)
	at org.elassandra.index.ElasticSecondaryIndex$ImmutableMappingInfo$SkinnyRowcumentIndexer$SkinnyRowcument.buildContext(ElasticSecondaryIndex.java:1435)
	at org.elassandra.index.ElasticSecondaryIndex$ImmutableMappingInfo$RowcumentIndexer$Rowcument.index(ElasticSecondaryIndex.java:1874)
	at org.elassandra.index.ElasticSecondaryIndex$ImmutableMappingInfo$RowcumentIndexer$Rowcument.index(ElasticSecondaryIndex.java:1863)
	at org.elassandra.index.ElasticSecondaryIndex$ImmutableMappingInfo$SkinnyRowcumentIndexer$SkinnyRowcument.index(ElasticSecondaryIndex.java:1435)
	at org.elassandra.index.ElasticSecondaryIndex$ImmutableMappingInfo$SkinnyRowcumentIndexer.flush(ElasticSecondaryIndex.java:1471)
	at org.elassandra.index.ElasticSecondaryIndex$ImmutableMappingInfo$RowcumentIndexer.finish(ElasticSecondaryIndex.java:1586)
	at org.apache.cassandra.index.SecondaryIndexManager$WriteTimeTransaction.commit(SecondaryIndexManager.java:983)
	at org.apache.cassandra.db.partitions.AtomicBTreePartition.addAllWithSizeDelta(AtomicBTreePartition.java:181)
	at org.apache.cassandra.db.Memtable.put(Memtable.java:284)
	at org.apache.cassandra.db.ColumnFamilyStore.apply(ColumnFamilyStore.java:1339)
	... 11 common frames omitted
```

**Comment by jxerome** [comment]
Works as expected with text fields.

```
$ curl -XPUT $search/elassandra -d '
> {
>   "mappings" : {
>     "fan" : {
>       "dynamic" : "strict",
>       "properties" : {
>         "name" : {
>           "type" : "text"
>         },
>         "like_elassandra" : {
>           "index" : false,
>           "type" : "text"
>         }
>       }
>     }
>   }
> }
> '
{"acknowledged":true,"shards_acknowledged":true}

$ curl -XPUT $search/elassandra/fan/jxerome -d '
> {
>     "name": "jxerome",
>     "like_elassandra" : "it is complicated"
> }
> '
{"_index":"elassandra","_type":"fan","_id":"jxerome","_version":1,"result":"created","_shards":{"total":1,"successful":1,"failed":0},"created":true}
```

**Comment by vroyer** [comment]
Hi Jerome,

Yes, there is bug in this case. The workaround is to set doc_values: false in the like_elassandra mapping. This will be fixed in the next release.
Thanks for reporting this issue.

**Comment by jxerome** [comment]
Hi Vincent,

The workaround works.
Give us time to wait until next release.
Thank you

**Comment by vroyer** [comment]
Hi Jerome,
Next release should be available next week.
The fix is available [there](https://github.com/strapdata/elassandra5-rc/commit/25ce120bb82bee32fda6bc2baac5aca308797a88) if needed.
Thanks.

**Comment by vroyer** [comment]
Fixed in release 5.5.0.6
